hid-ghlive/src/hid-ghlive.c

```diff
@@ -100,6 +115,46 @@ static void ghl_magic_poke(struct timer_list *t)
 	mod_timer(&sc->poke_timer, jiffies + GHL_GUITAR_POKE_INTERVAL*HZ);
 }
 
+static void ghl_magic_poke_ps4(struct timer_list *t)
+{
+	int ret;
+	struct urb *urb;
+	struct ghlive_sc *sc = from_timer(sc, t, poke_timer);
+	struct usb_device *usbdev = to_usb_device(sc->hdev->dev.parent->parent);
+	const u16 poke_size =
+		ARRAY_SIZE(ghl_ps4_magic_data);
+	unsigned int pipe = usb_sndctrlpipe(usbdev, 0x02);
```

Comment:
Should that be `unsigned int pipe = rcvctrlpipe(usbdev, 0x02);` ?

Reply:
Nevermind, that wouldn't make sense, it has to go OUT.

Reply:
If that helps, here's my device descriptor:
```
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x1430 RedOctane
  idProduct          0x07bb 
  bcdDevice            1.10
  iManufacturer           1 Activision
  iProduct                2 Guitar Hero 
  iSerial                 0 
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0029
    bNumInterfaces          1
    bConfigurationValue     1
    iConfiguration          0 
    bmAttributes         0xc0
      Self Powered
    MaxPower              500mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           2
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 
      bInterfaceProtocol      0 
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength     160
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               5
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x02  EP 2 OUT
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               5
```

Reply:
What are you running in order to see the bogus?

Reply:
Actually, i forgot to change the function from a Control to an Interrupt.

`unsigned int pipe = usb_sndctrlpipe(usbdev, 0x02);` should be `unsigned int pipe = usb_sndintpipe(usbdev, 0x02);`

Reply:
Indeed. That fixes it. Please include this change when you'll push the other things discussed on Slack.